### PR TITLE
feat(cdn.dedicated): add tab to see logs

### PR DESCRIPTION
**Pull request by tduboys**
### Description of the Change

Add log tab on dedicated cdn view using new API call /cdn/dedicated/{serviceName}/domains/{domain}/logs#POST

### Benefits

View real-time logs for a CDN offer

refer: DNS-1062, DNS-1080

### CODE REVIEW

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.service.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+angular.module('services').service('cdnDedicatedManageLogs', function (OvhHttp) {
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.service.js]: This is a deprecated way of doing things, please use https://github.com/ovh-ux/ovh-api-services instead :)

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.routes.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+angular.module('App').config(($stateProvider) => {
+  $stateProvider.state('app.networks.cdn.dedicated.logs', {
+    url: '/logs',
+    views: {
+      cdnView: {
+        templateUrl: 'cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html',
+        controller: 'CdnLogsCtrl',
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.routes.js]: Please use a `Component` instead :) here is an example: https://github.com/ovh-ux/ovh-manager-dedicated/blob/e4470a79acd07fbbac2f141a49a3498f8bc5d093/client/app/components/expiration/service-expiration-label.component.js

and the route would like like this:
```suggestion
         component: 'cndLogs',
```

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <ovh-tail-logs>
+        <div class="row ml-0"
+             data-ng-repeat="log in listCtrl.logger.logs track by $index"
+             data-ng-class="{
+                'text-primary' : log.message.status >= 200 && log.message.status < 300,
+                'text-warning' : log.message.status >= 300 && log.message.status < 499,
+                'text-danger'  : log.message.status > 499
+            }">
+
+            <div class="col-md-12">
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html]: No need for a
```js
<div class="row">
    <div class="col-md-12">
    </div>
</div>
```

A `div` is already as long as possible :)

You can write instead:
```js
<div
    class="ml-0"
    data-ng-repeat="...">
    <span ...></span>
</div>
```

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                <span data-ng-bind="log.message.source" title="ip_source" />
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html]: Avoid using `title`, it can't be seen by screen-readers and mobile devices

Comment (FredericEspiau) [reply client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html]: `span` require an ending tag `<span></span>`, even if it's rempty

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                <span title="ident">-</span>
+                <span data-ng-bind="log.message.user" title="user" />
+                [<span data-ng-bind="log.message.timestamp | date:'short'" title="date" />]
+                "<span data-ng-bind="log.message.method" title="method"/>
+                <span data-ng-if="log.message.is_ssl === 'NO'" title="schema">http://</span><span data-ng-if="log.message.is_ssl === 'YES'" title="schema">https://</span><span data-ng-bind="log.message.vhost" title="vhost"/><span data-ng-bind="log.message.uri" title="uri"/>
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html]: Why isn't `is_ssl` a boolean ?

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html (hunk 4 above)
Comment (FredericEspiau) [reply client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.html]: I understand what you are trying to do, you'd better use https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl instead :)

File: client/app/cdn/dedicated/domain/logs/cdn-dedicated-domain-logs.controller.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      });
+
+      this.startLog();
+    }
+
+    $onDestroy() {
+      this.logger.stop();
+    }
+
+    stopLog() {
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/domain/logs/cdn-dedicated-domain-logs.controller.js]: This isn't used

File: client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.controller.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      });
+
+      this.startLog();
+    }
+
+    $onDestroy() {
+      this.logger.stop();
+    }
+
+    stopLog() {
```
Comment (FredericEspiau) [inline client/app/cdn/dedicated/manage/logs/cdn-dedicated-manage-logs.controller.js]: This isn't used

**Comment by FredericEspiau** [comment]
Please target `develop` :)

**Comment by jleveugle** [comment]
Hello @tduboys,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**